nixos/modules/services/printing/cupsd.nix

```diff
@@ -259,8 +280,31 @@ in
           CUPSd temporary directory.
         '';
       };
-    };
 
+      startWhenNeeded = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Start CUPS processes on demand.
```

Comment:
Specifying expected extra latency on a specific piece of hardware would be useful. Something like 

`On my Intel-<specific model> this delays a printing dialog by 2.4 seconds`.